### ISSUE TITLE
can't build bundles because of calculating path to rxjs/CoreOperators.js

### ISSUE THREAD

**Issue by alexeagle** [angular/angular#7167]
release blocker. fyi @matsko 

Can't run `gulp build` because of rxjs beta.2 upgrade. @jeffbcross 
I think this is because CoreOperators.ts has no value-side https://github.com/ReactiveX/RxJS/blob/master/src/CoreOperators.ts
and system builder gets confused by empty files as indicated here
https://github.com/angular/angular/blob/master/gulpfile.js#L1151

**Comment by ericmartinezr** [comment]
fyi another issue reporting this https://github.com/angular/angular/issues/6929

**Comment by alexeagle** [comment]
let's keep the two issues separate

**Comment by alexeagle** [comment]
@jeffbcross I expected this to work:
https://github.com/angular/angular/commit/b9905017b713c71e8d8f4b479d803b0198d66a59

I don't know much about system bundler, can you take a look?

**Comment by jeffbcross** [comment]
@alexeagle I will take a look

**Comment by jeffbcross** [comment]
@alexeagle how can I verify this fix without the ts2dart fix?

**Comment by jeffbcross** [comment]
NM, I cherry-picked @vsavkin's fix from https://github.com/angular/angular/pull/7166/files

**Comment by alexeagle** [comment]
Update: the theory about this file being empty does not hold up. Jeff
experimented with adding an export statement but the system builder still
blows up.
We have no working theory about why we get this error or when it was
introduced.

**Comment by teropa** [comment]
@alexeagle @jeffbcross Managed to track this down. It’s basically got nothing to do with `rxjs/CoreOperators.js` per se. It’s got to do with incompatible versions of SystemJS.

SystemJS 0.18.10 uses a baseURL of `file:///path/to/angular`and the older 0.16.11 used by systemjs-builder uses `file:/path/to/angular`. For some reason when running a full build the builder is fed the incompatible `file:///path/to/angular` baseURL and it can’t find anything. The error does *not* occur if you only build '!bundle.js.dev’.

I don’t know exactly how the versions get mixed up, since when systemjs-loader requires systemjs it should get the version in its own node_modules. I suspect it has something to do with the fact that SystemJS shares instances of itself through `global`.

I was able to fix this by explicitly setting baseURL (see https://github.com/teropa/angular/commit/ab1a8040c2c3951827da49f2f64f255a7774cb6b). But I guess a proper fix would be to use versions of systemjs and systemjs-loader that play well with each other.

**Comment by alexeagle** [comment]
Thanks for tracking that down! Now we can release Angular again :)
@vsavkin is the current caretaker, I think we need to merge that fix to unblock the next release.

**Comment by teropa** [comment]
Alrighty, here's a PR in case you want it in :) https://github.com/angular/angular/pull/7360
